Sources/Sentry/SentryClient.m

```diff
@@ -130,13 +137,13 @@ - (SentryEvent *_Nullable)prepareEvent:(SentryEvent *)event
         event.dist = dist;
     }
 
+    // TODO fetzig: this might be the wrong place for beforeSend(...)
     if (nil != self.options.beforeSend) {
-        return self.options.beforeSend(event);
+        event = self.options.beforeSend(event);
     }
-    
-    return event;
-}
 
+    return [scope applyToEvent:event];
```

Comment:
`applyToEvent:event` must be called before `beforeSend` reason is that the user should get the fully built event before we send it.

Reply:
Moved `beforeSend` call to `SentryClient.captureEvent`.